api/app/Models/UserSkill.php

```diff
@@ -23,6 +23,26 @@ class UserSkill extends Model
 
     protected $keyType = 'string';
 
+    protected $fillable = [
+        'user_id',
+        'skill_id',
+        'skill_level',
+        'when_skill_used',
+    ];
+
+    /**
+     * model lifecycle methods
+     */
+    protected static function booted(): void
+    {
+        static::deleting(
+            function (UserSkill $userSkill) {
+                // soft delete all experience_skill records containing the model
+                ExperienceSkill::where('user_skill_id', $userSkill->attributes['id'])->delete();
```

Comment:
When working with a laravel Eloquent model, it's conventional to access attributes directly, instead of through the attributes array.
```suggestion
                ExperienceSkill::where('user_skill_id', $userSkill->id)->delete();
```

Reply:
Thanks, suggestion added in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7579/commits/4eefd5b3f5abcec389bd243de4e19295101a1be3
I can't remember where calling `->attributes` popped in first, I think from something I was playing around with 🤔